### DISCUSSION TITLE
How can check condition in query (react native firestore)

### DISCUSSION THREAD

**Discussion by vnxz** [invertase/react-native-firebase#5912] [Ideas]
I want to check condition in query data 
if auth user showmen true , in query where is ("showWome","==",true) if auth user showwomen true , in query where is("showMen","==",true) 
logic is male user who choose showWomen : true , to see only women user
and vise versa

how can check

```javascript
useEffect(() => {
  let unsub;
 
  const fetchCards = async()=>{

   
    const authgirlRef= query(collection(db,'users'),where("id","==",user.uid),where("showMen","==",true))
    const authgirl = await getDocs(authgirlRef)
    .then((snapshot)=>snapshot.docs.map(doc=>doc.id))
   
    const authboyRef= query(collection(db,'users'),where("id","==",user.uid),where("showWomen","==",true))
    const authboy = await getDocs(authboyRef)
    .then((snapshot)=>snapshot.docs.map(doc=>doc.id))

  

   


    const queryRef = collection(db,'users')

    unsub = onSnapshot(query(queryRef,where('showMen','==',true)
    ),
     snapshot=>{
       setProfiles(
         snapshot.docs
         .map(doc=>({
            id :doc.id,
            ...doc.data()
         })

         )
       )
    })
  }

  fetchCards();
  return unsub;

}, [])


```
can someone guide me

**Comment by mikehardy** [comment]
Hi there! Usage questions are best on stackoverflow - we try to focus on items that will result in changes in the module here, not just how to use the APIs. The community on stackoverflow is much larger and better able to help usage requests

**Reply by vnxz** [reply]
but this question need to view by experience react native developer , I already ask on stackoverflow, cannot answer the correct way.I think here will get answer

**Reply by mikehardy** [reply]
You might want to edit your question for clarity, I honestly could not figure out exactly what you wanted to do, what was happening now, vs what you wanted to happen in your question.
